paddle/fluid/inference/api/mkldnn_quantizer.cc

```diff
@@ -331,18 +339,23 @@ AnalysisPredictor::MkldnnQuantizer::GetMaxChScalingFactor(
         "Tensor is claimed to be unsigned, but its min value (%f) is < 0.0",
         min_val);
 
-  int channels = var_tensor.dims()[0];
-  LoDTensor scale_tensor = CreateScaleTensor(channels);
-  auto* scale_ptr = scale_tensor.mutable_data<double>(CPUPlace());
-
-  for (int i = 0; i < channels; ++i) {
-    const auto tensor = var_tensor.Slice(i, i + 1);
+  auto dims = var_tensor.dims();
+  bool is_transposed = oc_axis == (dims.size() - 1);
+  int num_col_dims = is_transposed ? oc_axis : oc_axis + 1;
```

Comment:
In that case, we would need to derive `oc_axis` again anyway so that it can be used here: https://github.com/PaddlePaddle/Paddle/pull/17641/files/904cd05467add542e690b5bd81f8e5ec78d2a7b4#diff-16a31361a76290e4220c11ea16e4e8f5R355

It would probably then something like 
```C++
int oc_axis = is_transposed;
```